### PR TITLE
Resource picker and display list on block

### PR DESCRIPTION
**Pull request by michaelatias**

### CODE REVIEW

File: client-side/src/app/block-editor/block-editor.component.ts [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
     @Output() hostEvents: EventEmitter<any> = new EventEmitter<any>();
 
-    constructor(private translate: TranslateService) {
+    constructor(private translate: TranslateService,
+                private blockEditorService: BlockEditorService
+               ) {
     }
 
     ngOnInit(): void {
+        this.blockEditorService.pluginUUID = "0e2ae61b-a26a-4c26-81fe-13bdd2e4aaa3"
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block-editor/block-editor.component.ts:23]: please change it. it should come from the hostObject.

File: client-side/src/app/block-editor/block-editor.component.ts [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
 import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { BlockEditorService } from './block-editor.service'
 
 @Component({
     selector: 'block-editor',
     templateUrl: './block-editor.component.html',
     styleUrls: ['./block-editor.component.scss']
 })
 export class BlockEditorComponent implements OnInit {
+    resourcesNames:  {key:string, value:string}[] = []
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block-editor/block-editor.component.ts:11]: there is a keyValuePair object in ngx-lib, see if you can use it instead

File: client-side/src/app/block-editor/block-editor.module.ts [hunk 3]
```diff
@@ -23,7 +26,8 @@ import { config } from '../addon.config';
     exports: [BlockEditorComponent],
     providers: [
         TranslateStore,
-        // Add here all used services.
+        BlockEditorService,
+        TranslateService
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block-editor/block-editor.module.ts:30]: why do you need the TranslateService in here? isn't the TranslateStore enough?

File: client-side/src/app/block/block.component.ts [hunk 4]
```diff
@@ -8,16 +11,93 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
 
+class DataSource implements IPepGenericListDataSource{
+    items: any[] = []
+    constructor(private translate: TranslateService, items: any[]){
+      this.items = items
+    }
+    setItems(items:any[]){
+      this.items = items
+    }
+    async init(params: { searchString?: string; filter?: any; sorting?: IPepListSortingChangeEvent; fromIndex: number; toIndex: number; }): Promise<IPepGenericListInitData> {
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block/block.component.ts:44]: don't you have a type to define? this is how Yaron is exposing it from the composite-lib?

File: client-side/src/app/block/block.component.ts [hunk 5]
```diff
@@ -8,16 +11,93 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+              Fields: [
+                {
+                  FieldID: 'Key',
+                  Type: 'TextBox',
+                  Title: this.translate.instant('Key'),
+                  Mandatory: false,
+                  ReadOnly: true
+                },
+                {
+                  FieldID: 'CreationDateTime',
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block/block.component.ts:63]: dates should be of type 'DateAndTime'

File: client-side/src/app/block/block.component.ts [hunk 6]
```diff
@@ -8,16 +11,93 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
...
+        private genericListService: PepGenericListService, private blockService: BlockService) {
     }
 
     ngOnInit(): void {
     }
-
     ngOnChanges(e: any): void {
+      if(this.hostObject.configuration){
+        this.resource = this.hostObject.configuration.resource
+        this.blockService.pluginUUID = "0e2ae61b-a26a-4c26-81fe-13bdd2e4aaa3"
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block/block.component.ts:28]: again, this should come from the hostObject

File: server-side/api.ts [hunk 7]
```diff
@@ -1,12 +1,21 @@
 import MyService from './my.service'
 import { Client, Request } from '@pepperi-addons/debug-server'
 
-// add functions here
-// this function will run on the 'api/foo' endpoint
-// the real function is runnning on another typescript file
-export async function foo(client: Client, request: Request) {
+export async function get_all_collections(client: Client, request: Request){
```
Comment (roiaharon-pepperi) [inline server-side/api.ts:4]: as of now, you are only displaying UDC collections. But in the future, this addon will serve all types of resources. try to create a more generic function? something like get_list_of_resources? and get the resource type as a query parameter

File: server-side/my.service.ts [hunk 8]
```diff
@@ -27,6 +27,14 @@ class MyService {
     getAddons(): Promise<InstalledAddon[]> {
         return this.papiClient.addons.installedAddons.find({});
     }
+    getAllUDCCollections(): Promise<any>{
```
Comment (roiaharon-pepperi) [inline server-side/my.service.ts:30]: you can use papi-sdk. there is a 'user_defined_collections' object. install the latest version

**Review by roiaharon-pepperi — CHANGES_REQUESTED** [review]

File: client-side/src/app/block-editor/block-editor.component.html [hunk 9]
```diff
@@ -1 +1,14 @@
-Block editor works...
\ No newline at end of file
+<mat-tab-group #tabgroutp class="editor-tabs">
```
Comment (roiaharon-pepperi) [inline client-side/src/app/block-editor/block-editor.component.html:1]: why do you need the tabs if you only have one tab?

File: client-side/src/app/block/block.component.ts (hunk 4 above)
Comment (michaelatias) [reply client-side/src/app/block/block.component.ts:44]: this is how Yaron exposing it

File: client-side/src/app/block-editor/block-editor.component.ts (hunk 1 above)
Comment (michaelatias) [reply client-side/src/app/block-editor/block-editor.component.ts:23]: the host object does not contains the addon uuid, i will put the addon uuid in constant on another file

File: client-side/src/app/block/block.component.ts (hunk 5 above)
Comment (michaelatias) [reply client-side/src/app/block/block.component.ts:63]: thats temporary code

File: client-side/src/app/block-editor/block-editor.component.html (hunk 9 above)
Comment (michaelatias) [reply client-side/src/app/block-editor/block-editor.component.html:1]: in the next sprint there will be another tab

File: client-side/src/app/block-editor/block-editor.component.ts (hunk 1 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/block-editor/block-editor.component.ts:23]: so I think that you should open a feature for tomer to add it. as it comes from the hostObject in the ATD Editor, it should come here as well

File: client-side/src/app/block/block.component.ts (hunk 4 above)
Comment (roiaharon-pepperi) [reply client-side/src/app/block/block.component.ts:44]: consider talking to him about it, this should be a type somewhere in the lib

**Pull request closed without merging**